client/ayon_houdini/plugins/create/create_usd_componentbuilder.py

```diff
@@ -0,0 +1,75 @@
+import inspect
+
+from ayon_houdini.api import plugin
+from ayon_core.pipeline import CreatedInstance, CreatorError
+
+import hou
+
+
+class CreateUSDComponentBuilder(plugin.HoudiniCreator):
+    identifier = "io.ayon.creators.houdini.componentbuilder"
+    label = "USD Component Builder LOPs"
+    product_type = "usd"
+    icon = "cubes"
+    description = "Create USD from Component Builder LOPs"
+
+    def get_detail_description(self):
+        return inspect.cleandoc("""
+            Creates a USD publish from a Component Output LOP that is part of 
+            a solaris component builder network.
+            
+            The created USD will contain the component builder LOPs and all its
+            dependencies inside the single product.
+            
+            To use it, select a Component Output LOP and click "Create" for
+            this creator. It will generate an instance for each selected
+            Component Output LOP.
+        """)
+
+    def create(self, product_name, instance_data, pre_create_data):
+
+        nodes = hou.selectedNodes()
+
+        builders = [
+            node for node in nodes if node.type().name() == "componentoutput"
```

Comment:
There must be a better way to ensure I'm exactly finding the right node type instead of only matching by name?

Reply:
I'm not sure I get it.
what are your the concerns about finding node by type definition name?

let me quote from Houdini [hou.NodeType class](https://www.sidefx.com/docs/houdini/hom/hou/NodeType.html)  docs: 
> All the node types in Houdini are organized into categories, and a node type is uniquely identified by its category and node type name.

Reply:
It's because the node type name itself can be used in other categories too. So to be entirely exact I **should** be checking against the category and the node type name. ;) So basically the current check isn't entirely explicit because it may pick up the wrong node types.